src/js/views/OperationsLog/Edit.jsx

```diff
@@ -0,0 +1,223 @@
+    const values = {
+      ...operationsLog,
+      recorded_at: ISO8601ToDatetimeLocal(operationsLog.recorded_at),
+      completed_at: operationsLog.completed_at
+        ? ISO8601ToDatetimeLocal(operationsLog.completed_at)
+        : operationsLog.completed_at
+    }
+
+    if (operationsLog.project_id) {
+      httpGet(
+        globalState.fetch,
+        new URL(`/projects/${operationsLog.project_id}`, globalState.baseURL),
+        (data) => {
+          values.project = {
+            project_id: data.id,
+            namespace_id: data.namespace_id,
+            project_type_id: data.project_type_id
+          }
+          setFieldValues(values)
+        },
+        (error) => onError(error)
+      )
+    } else {
+      setFieldValues(values)
+    }
+  }, [])
+
+  async function onSubmit() {
+    const url = new URL(
+      `/operations-log/${operationsLog.id}`,
+      globalState.baseURL
+    )
+    const newValues = {
+      ...fieldValues,
+      recorded_at: new Date(fieldValues.recorded_at).toISOString(),
+      completed_at: fieldValues.completed_at
+        ? new Date(fieldValues.completed_at).toISOString()
+        : null,
+      project_id: fieldValues.project_id ? fieldValues.project_id : null
+    }
+    delete newValues.project
+    const oldValues = {
+      ...operationsLog,
+      recorded_at: new Date(operationsLog.recorded_at).toISOString(),
+      completed_at: operationsLog.completed_at
+        ? new Date(operationsLog.completed_at).toISOString()
+        : null
+    }
+
+    const patchValue = compare(oldValues, newValues)
+    if (patchValue.length === 0) {
+      onCancel()
+      return
+    }
+    const response = await httpPatch(globalState.fetch, url, patchValue)
+    if (response.success) {
+      onSuccess()
+    } else {
+      onError(response.data)
+    }
+  }
+
+  function onValueChange(key, value) {
+    setFieldValues((prevValues) => {
+      const newValues = { ...prevValues }
+      if (key === 'project') {
+        delete newValues.project
+        newValues.project_id = value
+      } else {
+        newValues[key] = value
+      }
+      return newValues
+    })
+  }
+
+  if (!fieldValues) return <></>
+
+  return (
+    <Form.SimpleForm
+      errorMessage={null}
+      onCancel={onCancel}
+      onSubmit={onSubmit}
+      ready={true}
+      saving={saving}>
+      <Form.Field
+        title={t('operationsLog.changeType')}
+        name="change_type"
+        type="select"
+        options={[
+          { label: 'Configured', value: 'Configured' },
+          { label: 'Decommissioned', value: 'Decommissioned' },
+          { label: 'Deployed', value: 'Deployed' },
+          { label: 'Migrated', value: 'Migrated' },
+          { label: 'Provisioned', value: 'Provisioned' },
+          { label: 'Restarted', value: 'Restarted' },
+          { label: 'Rolled Back', value: 'Rolled Back' },
+          { label: 'Scaled', value: 'Scaled' },
+          { label: 'Upgraded', value: 'Upgraded' }
```

Comment:
Should [the jsonschema](https://github.com/AWeber-Imbi/imbi-ui/pull/19/files#diff-026470013a7b33c5d9975fb976a7ccec15b881e62441bf009c121f91b7f48a96R17) be enforcing these as enum values?

Reply:
Actually these might be best exposed as fields in settings alongside `projectFactTypes`, `projectLinkTypes`, et.al.

Reply:
I agree, in settings but still hardcoded is a good compromise. That way, it at least won't be duplicated everywhere, and it's where you'd expect it to be (settings metadata)